src/error/mod.rs

```diff
@@ -0,0 +1,139 @@
+// This file is part of Twig (ported to Rust).
+//
+// For the copyright and license information, please view the LICENSE
+// file that was distributed with this source code.
+
+/// Twig generic error
+
+use std::fmt::{self, Display};
+
+
+#[macro_use]
+pub mod macros;
+// use std Error-trait to improve cross-crate compatibility
+// don't mix it up with Err(X)
+pub use std::error;
+pub mod api;
+
+// generic wrapper around some ErrorCode - adds location support
+#[derive(Debug)]
+pub struct Error<T>
```

Comment:
Maybe call `ErrorCode` -> `TwigError`? I understand that we are running of `Error` names here a bit :)